### PR TITLE
Fixed Value of created_at and updated_at columns not updating in ui_bookmark table

### LINKED ISSUE #18557: Value of created_at and updated_at columns not updating in ui_bookmark table
### Preconditions (*)

Magento 2.2.5

### Steps to reproduce (*)

Value of **created_at** and **updated_at** columns not updating in **ui_bookmark** table.
Screenshot
expected-result

### Expected result (*)

1. Column **created_at** and **updated_at** should be update.

### Actual result (*)

1. Column **created_at** and **updated_at** not updating.

### PR DESCRIPTION
**Pull request by shikhamis11**
Fixed #18557 for Magento 2.3
Value of created_at and updated_at columns not updating in ui_bookmark table

**Preconditions (*)**
Magento 2.2.5 and above

**Steps to reproduce (*)**
Value of created_at and updated_at columns not updating in ui_bookmark table.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3704](https://jira.corp.magento.com/browse/ENGCOM-3704) has been created to process this Pull Request

**Comment by shikhamis11** [comment]
@magento-engcom-team and @orlangur it's been a month this pull request is in review in progress .. is it going to be merged or not ?

**Comment by sivaschenko** [comment]
Hi @shikhamis11 , sorry for the delay in response. There is a problem with the pull request discovered during testing: 'created_at' field value of bookmark is getting updated together with updated_at. Can you please take a look at that?

**Comment by shikhamis11** [comment]
sure I will check

**Comment by shikhamis11** [comment]
@sivaschenko Please check now this seems working on manual testing scenario

**Comment by sivaschenko** [comment]
@shikhamis11 thank you, we'll retest the changes

File: app/code/Magento/Ui/etc/db_schema.xml [hunk 1]
```diff
@@ -18,8 +18,10 @@
                 comment="Mark current bookmark per user and identifier"/>
         <column xsi:type="varchar" name="title" nullable="true" length="255" comment="Bookmark title"/>
         <column xsi:type="longtext" name="config" nullable="true" comment="Bookmark config"/>
-        <column xsi:type="datetime" name="created_at" on_update="false" nullable="false" comment="Bookmark created at"/>
-        <column xsi:type="datetime" name="updated_at" on_update="false" nullable="false" comment="Bookmark updated at"/>
+        <column xsi:type="timestamp" name="created_at" on_update="false" nullable="false" default="CURRENT_TIMESTAMP"
```
Comment (sidolov) [inline app/code/Magento/Ui/etc/db_schema.xml]: Unfortunately, we are not allowed to change column type according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#db-schema)

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/etc/db_schema.xml (hunk 1 above)
Comment (shikhamis11) [reply app/code/Magento/Ui/etc/db_schema.xml]: @sidolov ok I will check if it is possible to fix the issue without changing the column type

**Comment by sidolov** [comment]
@shikhamis11 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by orlangur** [comment]
@sidolov can we change default value and not changing type? It's possible since MySQL 5.6.5: https://dev.mysql.com/doc/refman/5.6/en/timestamp-initialization.html

I see that guide is saying changing default value is prohibited but current one is buggy and it also makes no sense if we immediately rewrite value on application level: https://github.com/magento/magento2/pull/22145/files

**Comment by sidolov** [comment]
Hi @shikhamis11 , @orlangur recently we discussed such case with our architect and decided to accept current PR as reasonable. 

We need additionally verify a few cases to make sure that the solution works properly.

**Notes for QA**
Please, verify the following cases with provided changes: 
- The dates are saved in GMT/proper timezone, following the best practices
- The dates are verified when shown on front-end/ backend to be properly converted to the right timezone/locale.
- Testing should take into account server TZ, Mysql TZ, PHP TZ, TZ set in admin and the user locale/TZ for frontend.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4742](https://jira.corp.magento.com/browse/ENGCOM-4742) has been created to process this Pull Request

**Review by orlangur — CHANGES_REQUESTED** [review]
@sidolov wait a moment, column type must not be changed, just default value added to column opf type `datetime`.

**Comment by shikhamis11** [comment]
@orlangur should I change the column type `datetime` instead of `timestamp` ?

**Comment by orlangur** [comment]
@shikhamis11 yep, just revert all changes but a new default value and squash into a single commit.

**Comment by sidolov** [comment]
Hi @orlangur , you are right, missed column type change. Thank you!
@shikhamis11, please, revert column type change and leave only default type modification

**Comment by shikhamis11** [comment]
@orlangur @sidolov I have a doubt 
should I need to change the default value also .. I don't find any specific default value for `datetime` field. 
I think `CURRENT_TIMESTAMP` will work here. Please let me know your opinion about this.

**Comment by orlangur** [comment]
@shikhamis11, I meant that default value, see

> can we change default value and not changing type? It's possible since MySQL 5.6.5: https://dev.mysql.com/doc/refman/5.6/en/timestamp-initialization.html

**Comment by shikhamis11** [comment]
@orlangur while force push this Pr is closed .. I have created new Please review that 
#22340

**Comment by orlangur** [comment]
@shikhamis11 it means that you pushed it without a needed commit. Ok.

**Pull request closed without merging**